builtin/providers/azure/utils.go

```diff
@@ -0,0 +1,37 @@
+package azure
+
+import (
+	"math/rand"
+	"strings"
+)
+
+// reverseDNSName simply reverses the provided DNS name.
+func reverseDNSName(dnsName string) string {
+	bits := strings.Split(dnsName, ".")
+	// reverse the bits:
+	for i, j := 0, len(bits)-1; i < j; i, j = i+1, j-1 {
+		bits[i], bits[j] = bits[j], bits[i]
+	}
+	return strings.Join(bits, ".")
+}
+
+// getRandomStringLabel returns a random string of the given length.
+func getRandomStringLabel(n int) string {
+	var chars = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+	buf := make([]rune, n)
+	for i := 0; i < n; i++ {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return string(buf)
+}
```

Comment:
We actually have a helper for this functionality already: [`resource.UniqueId()`](https://github.com/hashicorp/terraform/blob/cb36bacf8eaa1f66d752314eb5a3e718b283e877/helper/resource/id.go#L17-L23)

Think you could switch to using that? If the places where you pass in length `20` are truly length constrained, we can talk about extending it with another helper function that accepts length like yours does.